internal/command/test.go

```diff
@@ -1009,10 +1009,10 @@ func (runner *TestFileRunner) Cleanup(file *moduletest.File) {
 		states = append(states, state)
 	}
 
-	slices.SortFunc(states, func(a, b *TestFileState) bool {
+	slices.SortFunc(states, func(a, b *TestFileState) int {
 		// We want to clean up later run blocks first. So, we'll sort this in
 		// reverse according to index. This means larger indices first.
-		return a.Run.Index > b.Run.Index
+		return b.Run.Index - a.Run.Index
```

Comment:
`aws-sdk-go-base/v2` uses the latest version of `golang.org/x/exp`, which includes a breaking change to the [`slices.SortFunc`](https://pkg.go.dev/golang.org/x/exp@v0.0.0-20230817173708-d852ddb80c63/slices#SortFunc) signature:


```
func SortFunc[S ~[]E, E any](x S, cmp func(a, b E) int)
```

> SortFunc sorts the slice x in ascending order as determined by the cmp function. This sort is not guaranteed to be stable. cmp(a, b) should return a negative number when a < b, a positive number when a > b and zero when a == b.

To continue sorting in descending order the arguments were switched (b - a). A simple sanity is included below, for reference.

```go
// Old signature
func main() {
	sl := []int{5, 1, 3, 4, 2, 6}
	fmt.Printf("before: %v\n", sl)
	// before: [5 1 3 4 2 6]

	slices.SortFunc(sl, func(a, b int) bool {
		return a > b
	})
	fmt.Printf("after: %v\n", sl)
	// after: [6 5 4 3 2 1]
}
```

```go
// New signature
func main() {
	sl := []int{5, 1, 3, 4, 2, 6}
	fmt.Printf("before: %v\n", sl)
	// before: [5 1 3 4 2 6]

	slices.SortFunc(sl, func(a, b int) int {
		return b - a
	})
	fmt.Printf("after: %v\n", sl)
	// after: [6 5 4 3 2 1]
}
```

Lastly, this appears to be the only place the `x/exp` package is utilized outside of the S3 backend, so there should be no impact elsewhere.

```console
% rg "golang.org/x/exp" -g '!/go.*'
internal/command/test.go
17:     "golang.org/x/exp/slices"

internal/backend/remote-state/s3/backend_test.go
34:     "golang.org/x/exp/maps"
```